odoo/addons/base/models/ir_attachment.py

```diff
@@ -511,13 +511,16 @@ def create(self, vals_list):
     def _post_add_create(self):
         pass
 
-    @api.one
     def generate_access_token(self):
-        if self.access_token:
-            return self.access_token
-        access_token = self._generate_access_token()
-        self.write({'access_token': access_token})
-        return access_token
+        valses = []
+        for attachment in self:
+            if attachment.access_token:
+                valses.append(attachment.access_token)
+                continue
+            access_token = self._generate_access_token()
```

Comment:
I wonder, `_generate_access_token` may depend on its `self`
```suggestion
            access_token = attachment._generate_access_token()
```

Reply:
I don't think so, the definition is below, it's simply `return str(uuid.uuid4())` and in fact it was inlined back when I wrote the original PR (I know because it conflicted when rebasing)